sdk/src/nonce_instruction.rs

```diff
@@ -63,12 +63,25 @@ pub enum NonceInstruction {
     ///     0 - A NonceAccount in the Uninitialized state
     ///     1 - RecentBlockHashes sysvar
     ///     2 - Rent sysvar
-    Initialize,
+    ///
+    /// The `Pubkey` parameter specifies the entity authorized to execute nonce
+    /// instruction on the account
+    Initialize(Pubkey),
+
+    /// `Authorize` changes the entity authorized to execute nonce instructions
+    /// on the account
+    ///
```

Comment:
Can you indicate who needs to sign the transaction here.  I assume it's the current authorized pubkey